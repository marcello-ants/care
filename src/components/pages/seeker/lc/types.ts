import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
