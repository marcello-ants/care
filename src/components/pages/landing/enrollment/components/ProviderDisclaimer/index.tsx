// External Dependencies
import { Typography, Link } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

export default function ProviderDisclaimer() {
  // Styles
  const classes = useStyles();

  return (
    <Typography careVariant="disclaimer1" className={classes.disclaimer}>
      <span>
        You must be 18 years or older to use Care.com. By clicking &quot;Submit,&quot; you agree to
        our
      </span>{' '}
      <Link href="/terms-of-use-p1012.html" careVariant="disclaimer1">
        Terms of Use
      </Link>{' '}
      <span>and</span>{' '}
      <Link href="/privacy-policy-p1013.html" careVariant="disclaimer1">
        Privacy Policy
      </Link>
      <span>.</span>
    </Typography>
  );
}
