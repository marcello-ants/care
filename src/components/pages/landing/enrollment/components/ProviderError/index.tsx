// External Dependencies
import { Box } from '@material-ui/core';
import { Icon24UtilityWarning } from '@care/react-icons';
import { Typography } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

function ProviderError() {
  // Styles
  const classes = useStyles();

  return (
    <Box className={classes.errorBanner}>
      <Box>
        <Icon24UtilityWarning color="#CF230C" />
      </Box>
      <Box>
        <Typography variant="h4">Whoops! Something went wrong</Typography>
        <Typography careVariant="body3">
          There was an error when we tried to submit your request. Please try again.
        </Typography>
      </Box>
    </Box>
  );
}

export default ProviderError;
