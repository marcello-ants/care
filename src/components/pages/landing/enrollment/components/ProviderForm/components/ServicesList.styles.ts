import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  empty: {
    '& > .MuiSelect-select': {
      color: theme.palette.care?.grey[700],
      fontWeight: 'normal',
    },
  },
  inputContainer: {
    width: '100%',
    '& > div': {
      padding: 0,
    },
  },
  error: {
    color: theme?.palette?.care?.red[800],
    marginLeft: theme.spacing(1.25),
    marginTop: theme.spacing(0.5),
  },
}));

export default useStyles;
