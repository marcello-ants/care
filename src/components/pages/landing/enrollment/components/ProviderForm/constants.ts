export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
