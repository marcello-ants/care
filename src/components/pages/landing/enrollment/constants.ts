export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
