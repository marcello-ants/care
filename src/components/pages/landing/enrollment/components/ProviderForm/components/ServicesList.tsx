/* eslint-disable import/prefer-default-export */
// External Dependencies
import { useState } from 'react';
import { Select, MenuItem, useMediaQuery } from '@material-ui/core';
import { useFormikContext } from 'formik';
import { theme } from '@care/material-ui-theme';
import { Typography } from '@care/react-component-lib';

// Internal Constants
import { PROVIDER_SERVICES_DESKTOP, PROVIDER_SERVICES_MOBILE } from '../constants';

// Styles
import useStyles from './ServicesList.styles';

function ServicesList() {
  // Styles
  const classes = useStyles();
  const isDesktop = useMediaQuery(theme.breakpoints.up('md'));

  const formik = useFormikContext<{ providerService: string }>();
  const listOfServices = isDesktop ? PROVIDER_SERVICES_DESKTOP : PROVIDER_SERVICES_MOBILE;

  // Component's State
  const [typeOfService, setTypeOfService] = useState('');

  // Handlers
  const onServiceValueChange = (value: string) => {
    formik.setFieldValue('providerService', value);
    setTypeOfService(value as string);
  };

  return (
    <>
      <Select
        name="selectJobType"
        className={
          !typeOfService
            ? `${classes.inputContainer} ${classes.empty}`
            : `${classes.inputContainer}`
        }
        value={typeOfService}
        displayEmpty
        onChange={(e) => {
          onServiceValueChange(e.target.value as string);
        }}>
        <MenuItem value="" disabled>
          Select one:
        </MenuItem>
        {listOfServices.map((service: any) => {
          return (
            <MenuItem value={service.value} key={service.value}>
              {service.label}
            </MenuItem>
          );
        })}
      </Select>
      {formik.errors.providerService && (
        <Typography careVariant="info1" className={classes.error}>
          Please select a job
        </Typography>
      )}
    </>
  );
}

export { ServicesList };
