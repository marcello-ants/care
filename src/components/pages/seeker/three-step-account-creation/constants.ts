import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
