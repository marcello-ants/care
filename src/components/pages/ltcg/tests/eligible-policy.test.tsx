import { screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { LTCG_ROUTES } from '@/constants';
import { useAppDispatch } from '@/components/AppState';
import { preRenderPage } from '@/__setup__/testUtil';
import { YesOrNoAnswer } from '@/__generated__/globalTypes';
import { initialAppState } from '@/state';
import { AppState } from '@/types/app';
import EligiblePolicy from '../eligible-policy';

jest.mock('@/components/AppState', () => ({
  ...(jest.requireActual('@/components/AppState') as object),
  useAppDispatch: jest.fn(),
}));

describe('EligiblePolicy', () => {
  it('matches snapshot', () => {
    const { renderFn } = preRenderPage();
    const { asFragment } = renderFn(EligiblePolicy);
    expect(asFragment()).toMatchSnapshot();
  });

  it('navigate to "details about yourself" page when "No" is clicked', async () => {
    const mockAppDispatch: ReturnType<typeof useAppDispatch> = jest.fn();
    (useAppDispatch as jest.Mock).mockReturnValue(mockAppDispatch);

    const { renderFn, routerMock } = preRenderPage();
    renderFn(EligiblePolicy);

    const noButton = screen.getByText('No');
    fireEvent.click(noButton);
    await waitFor(() =>
      expect(routerMock.push).toHaveBeenCalledWith(LTCG_ROUTES.DETAILS_ABOUT_YOURSELF)
    );

    await waitFor(() =>
      expect(mockAppDispatch).toHaveBeenCalledWith({
        type: 'setLtcgCaregiverNeeded',
        caregiverNeeded: YesOrNoAnswer.NO,
      })
    );
  });

  it('navigate to the "where" page when "Yes" is clicked', async () => {
    const { renderFn, routerMock } = preRenderPage();
    renderFn(EligiblePolicy);

    const yesButton = screen.getByText('Yes');
    userEvent.click(yesButton);
    await waitFor(() => expect(routerMock.push).toHaveBeenCalledWith(LTCG_ROUTES.WHERE));
  });

  it('should pre-select the option stored in app state', async () => {
    const appState: AppState = {
      ...initialAppState,
      ltcg: {
        ...initialAppState.ltcg,
        caregiverNeeded: YesOrNoAnswer.NO,
      },
    };
    const { renderFn } = preRenderPage({ appState });
    renderFn(EligiblePolicy);

    const noButton = screen.getByDisplayValue('NO');
    expect(noButton).toBeChecked();
  });
});
