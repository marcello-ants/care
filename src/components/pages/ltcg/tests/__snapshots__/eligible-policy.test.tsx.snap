// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EligiblePolicy matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-iconContainer-2"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-root-4"
      focusable="false"
      viewBox="0 0 64 64"
    >
      <circle
        cx="32"
        cy="32"
        fill="#00a779"
        r="26.7"
      />
      <path
        d="M28.65 39.88a1.1 1.1 0 01-.83-.38L22 32.94a1.12 1.12 0 01.09-1.58 1.13 1.13 0 011.59.09l5 5.64 11.7-12.61A1.12 1.12 0 1142 26L29.47 39.52a1.11 1.11 0 01-.82.36z"
        fill="#fff"
      />
      <path
        d="M28.65 40.24a1.49 1.49 0 01-1.09-.5l-5.84-6.56a1.4 1.4 0 01-.37-1.07 1.42 1.42 0 01.49-1 1.46 1.46 0 012.08.12l4.76 5.34 11.43-12.33a1.42 1.42 0 011-.47 1.44 1.44 0 011.06.39 1.47 1.47 0 01.08 2.09L29.74 39.76a1.51 1.51 0 01-1.09.48zm-5.84-8.81a.78.78 0 00-.51.2.74.74 0 00-.25.53.7.7 0 00.19.55l5.84 6.56a.75.75 0 00.56.25.78.78 0 00.56-.24l12.55-13.52a.76.76 0 000-1.08.71.71 0 00-.55-.2.77.77 0 00-.53.24l-12 12.9-5.28-5.93a.76.76 0 00-.58-.26z"
        fill="#fff"
      />
    </svg>
  </div>
  <h2
    class="MuiTypography-root makeStyles-header-5 MuiTypography-h2"
  >
    Good news! Your policy is eligible for use with Care.com.
  </h2>
  <h4
    class="MuiTypography-root makeStyles-pillTitle-1 MuiTypography-h4"
  >
    Do you need to find a caregiver?
  </h4>
  <ul
    class="MuiList-root makeStyles-selector-3 MuiList-padding"
  >
    <li
      class="MuiListItem-root makeStyles-root-7 pill MuiListItem-gutters"
    >
      <input
        class="makeStyles-input-10"
        id="selector.undefined.0"
        name=""
        readonly=""
        type="radio"
        value="YES"
      />
      <div
        class="makeStyles-controlIndicator-11 md"
        tabindex="0"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
        >
          <div>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Yes
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="MuiListItem-root makeStyles-root-7 pill MuiListItem-gutters"
    >
      <input
        class="makeStyles-input-10"
        id="selector.undefined.1"
        name=""
        readonly=""
        type="radio"
        value="NO"
      />
      <div
        class="makeStyles-controlIndicator-11 md"
        tabindex="0"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
        >
          <div>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              No
            </p>
          </div>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
